### PR TITLE
Bump node version to node20 and update dependencies

### PR DESCRIPTION
**Pull request by watsonian**
This moves the action from `node16` to `node20` in preparation for [GitHub moving all all actions to Node20](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/). This also updates all dependencies.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cli-action",
-  "version": "2.0.0",
+  "version": "3.0.0",
```
Comment (nmanoogian) [inline package.json:3]: check: Do we need to bump the major version? I would have thought this would be a patch update (e.g. v3.0.1)

**Review by nmanoogian — APPROVED** [review]
Thanks for picking this up, @watsonian!

Just confirming, did you get a chance to test this?

File: package.json (hunk 1 above)
Comment (watsonian) [reply package.json:3]: Is it common to use point release versioning for this?

```
uses: dopplerhq/cli-action@v2
```

I'm not overly familiar with it, but have mostly seen full version bumps. From a change perspective, a point release makes sense for sure.

**Comment by watsonian** [comment]
> Just confirming, did you get a chance to test this?

I didn't test it myself, but the tests for the action run through the installation process and verify the CLI got installed as expected. Are we able to have a workflow from a separate project that installs an unpublished action by SHA or something? I could do that if we don't think the tests are sufficient.

**Comment by patsissons** [comment]
I ran this through our CI and seemed to work great 🎉 

```diff
-        uses: dopplerhq/cli-action@v2
+        uses: dopplerhq/cli-action@watsonian/bump-nodejs-version
```

File: package.json (hunk 1 above)
Comment (watsonian) [reply package.json:3]: Looks like it's just using branch or tag names, so we could definitely release a point release if we want to do something like `v2.0.1`

**Comment by nmanoogian** [comment]
> I ran this through our CI and seemed to work great 🎉

Thanks @patsissons!

**Pull request merged**